core/ir/synthetic/encoding_common.h

```diff
@@ -0,0 +1,147 @@
+/* **********************************************************
+ * Copyright (c) 2024 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#ifndef _SYNTHETIC_ENCODING_COMMON_H_
+#define _SYNTHETIC_ENCODING_COMMON_H_
+
+/**
+ * This synthetic ISA (which we call DR_ISA_REGDEPS) is a made up ISA that has the purpose
+ * of preserving register dependencies and giving hints on the type of operations each
+ * instruction is performing.
+ * For this reason the majority of operations that would normally work on instructions
+ * coming from an actual ISA (e.g., DR_ISA_AMD64) are not supported.
+ * We support instr_convert_to_isa_regdeps(), which converts an #instr_t of an actual ISA
+ * to a DR_ISA_REGDEPS instruction.
+ * Other operations we support (related to encoding and decoding of a DR_ISA_REGDEPS
+ * #instr_t) are: instr_encode(), instr_encode_to_copy(), decode(), decode_common().
+ * DR_ISA_REGDEPS #instr_t can only return the encoded information described in the
```

Comment:
See the PR description comment: I find this formatting hard to read with every line starting flush left.  Blank lines or indentations on the *left* side to delineate paragraphs will help.

Reply:
Done. https://github.com/DynamoRIO/dynamorio/pull/6691#discussion_r1554658906